tests/functional/test_location_change.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+import os
+from dbt.tests.util import run_dbt
+
+_MODEL_SQL = """
+select 1 as id
+"""
+
+_INVALID_LOCATION = os.getenv('DBT_TEST_BIGQUERY_BAD_LOCATION', 'northamerica-northeast1')
+_VALID_LOCATION = os.getenv('DBT_TEST_BIGQUERY_INITIAL_LOCATION', 'US')
+
+class BaseBigQueryLocation:
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "model.sql": _MODEL_SQL,
+        }
+
+
+class TestBigqueryValidLocation(BaseBigQueryLocation):
+
+    def test_bigquery_valid_location(self, project):
+        results = run_dbt()
+        for result in results:
+            assert "US" == result.adapter_response["location"]
+
+
+class TestBigqueryInvalidLocation(BaseBigQueryLocation):
+
+    @pytest.fixture(scope="class")
+    def profiles_config_update(self, dbt_profile_target):
+        outputs = {"default": dbt_profile_target}
+        outputs["default"]["location"] = _INVALID_LOCATION
+        yield
+        outputs = {"default": dbt_profile_target}
+        outputs["default"]["location"] = _VALID_LOCATION
```

Comment:
Would be amazing if you could:
```
try:
finally:
```
which would guarantee except in the case the code fails so bad it errs out completely that we flip this back. Unsure though if that paradigm plays nicely with `yield` in pytest. Maybe something to Google?

Reply:
not sure I follow where/why we want try...finally here, can you expand on this a bit?

Reply:
I believe the worry is that if this tests fails during run once we can fuse finally to force it to try and reset the variable again back to default to not break the next test

Reply:
I believe the `yield` functionality as managed by pytest guarantees that no matter the result of the test associated with the fixture the code after will be executed